content/en/docs/hertz/tutorials/toolkit/template.md

```diff
@@ -0,0 +1,473 @@
+```
+
+### render data:
+
+```
+{
+  "*": {
+    "GoModule": "github.com/hertz/hello",
+    "ServiceName": "hello",
+    "UseApacheThrift": true
+  },
+  "main.go": {
+    "MainData": "this is customized render data"
+  }
+}
+```
+Command:
+```
+hz new --mod=github.com/hertz/hello --idl=./hertzDemo/hello.thrift --customize_layout=template/layout.yaml:template/data.json
+```
+
+# Custom package template
+
+> The template address of the hz template:
+>
+> Users can modify or rewrite according to the default template to meet their own needs
+
+-   The so-called package template refers to the code related to the idl definition service. This part of the code involves the service, go_package/namespace, etc. specified when defining idl. It mainly includes the following parts:
+-   handler.go: Handling function template logic
+-   router.go: the route registration logic of the service defined by the specific idl
+-   register.go: call the logic of the content in the router.go
+-   ~~Model code: generated go struct; however, since the tool that uses plugins to generate model code currently does not have permission to modify the model template, this part of the function is not open for now~~
+
+## Command
+
+```
+# After that, the package template rendering data will be provided, so the form of "k-v" is retained when entering the command, and ":" needs to be added after customize_package. 
+hz new --mod=github.com/hertz/hello --handler_dir=handler_test --idl=hertzDemo/hello.thrift --customize_package=template/package.yaml:
+```
+
+## Default package template
+
+Note: The custom package template function does not give rendering data. This is mainly because the rendering data is parsed and generated by the hz tool, so the function of writing your own rendering data is not provided for the time being. You can modify the parts of the template that are not related to the rendering data to meet your own needs.
+
+```
+# The following data is obtained by yaml marshal, so it may look messy 
+layouts:
+  # Path only represents handler.go template, the specific handler path is determined by the default path and handler_dir 
+  - path: handler.go
+    delims:
+      - '{{'
+      - '}}'
+    body: "// Code generated by hertz generator.\n\npackage {{.PackageName}}\n\nimport (\n\t\"context\"\n\n\t\"github.com/cloudwego/hertz/pkg/app\"\n\n
+    {{- range $k, $v := .Imports}}\n\t{{$k}} \"{{$v.Package}}\"\n{{- end}}\n)\n\n{{range $_, $MethodInfo := .Methods}}\n{{$MethodInfo.Comment}}\n
+    func {{$MethodInfo.Name}}(ctx context.Context, c *app.RequestContext) { \n\tvar err error\n\t{{if ne $MethodInfo.RequestTypeName \"\" -}}\n\t
+    var req {{$MethodInfo.RequestTypeName}}\n\terr = c.BindAndValidate(&req)\n\tif err != nil {\n\t\tc.String(400, err.Error())\n\t\treturn\n\t}\n\t{{end}}\n
+    \tresp := new({{$MethodInfo.ReturnTypeName}})\n\n\tc.{{.Serializer}}(200, resp)\n}\n{{end}}\n\t\t\t"
+  # path只表示router.go的模板，其路径固定在：biz/router/namespace/
```

Comment:
Path only represents the router.go template, whose path is fixed at: biz/router/namespace.

Reply:
done